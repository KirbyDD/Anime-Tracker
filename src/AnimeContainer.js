import React, { Component } from 'react'

class AnimeContainer extends Component {
  constructor() {
    super();
    this.state = {
      current: false
    }
  }

  render() {
    return (
      null
    )
  }
}

export default AnimeContainer;