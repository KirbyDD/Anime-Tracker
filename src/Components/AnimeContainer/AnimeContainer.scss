.list-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}

.shows-list {
  text-align: center;
  font-size: 48px;
}

.buttons {
  display: flex;
  justify-content: center;
  margin: 15px;
  button {
    color: gold;
    background-color: black;
    box-shadow: 7px 7px 9px;
    font-size: 1.5rem;
    height: 60px;
    margin: 2px 20px;
    width: 160px;
    &:hover {
      box-shadow: 3px 5px 3px white;
      color: white;
      transition: .3s;
    }
  }
}