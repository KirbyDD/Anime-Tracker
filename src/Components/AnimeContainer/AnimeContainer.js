import React from 'react';
import AnimeSnippet from '../AnimeSnippet/AnimeSnippet'
import {connect} from 'react-redux';
import './AnimeContainer.scss'

export const AnimeContainer = (props) => {
  let list = props.anime.map(anime => {
    return <AnimeSnippet {...anime} key={anime.id}/>
  })
  return (
    <section>
      <h2>List of Shows</h2>
      <div className="list-container">
        {list}
      </div>
    </section>
  )
}

const mapStateToProps = store => ({
  anime: store.anime
})

export default connect(mapStateToProps)(AnimeContainer);