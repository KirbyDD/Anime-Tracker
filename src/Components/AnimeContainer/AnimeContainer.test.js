import React from "react";
import { shallow } from "enzyme";
import { AnimeContainer } from "./AnimeContainer";

let anime = [
  {
    name: "anime1",
    id: 1
  },
  {
    name: "anime2",
    id: 2
  },
  {
    name: "anime3",
    id: 3
  }
];

describe("Anime Container", () => {
  let wrapper;

  beforeEach(function() {
    wrapper = shallow(<AnimeContainer anime={anime} />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
