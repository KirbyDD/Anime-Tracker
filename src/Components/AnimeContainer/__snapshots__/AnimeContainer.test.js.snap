// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anime Container should match snapshot 1`] = `
<section>
  <h2>
    List of Shows
  </h2>
  <div
    className="list-container"
  >
    <AnimeSnippet
      id={1}
      key="1"
      name="anime1"
    />
    <AnimeSnippet
      id={2}
      key="2"
      name="anime2"
    />
    <AnimeSnippet
      id={3}
      key="3"
      name="anime3"
    />
  </div>
</section>
`;
